src/js/popup/popup.js

```diff
@@ -309,9 +314,44 @@
           
           const generateRandomMask = document.querySelector(".js-generate-random-mask");
           const { premium } = await browser.storage.local.get("premium");
+          const { dataCollection } = await browser.storage.local.get(
+            "dataCollection"
+          );
           const maskPanel = document.getElementById("masks-panel");
           let getMasksOptions = { fetchCustomMasks: false };
-          
+
+          // logic to show survey is found in shouldShowSurvey function
+          const shouldShowCSAT = await popup.panel.survey.utils.shouldShowSurvey();
+
+          if (shouldShowCSAT && dataCollection === "data-enabled") {
+            const survey = popup.panel.survey;
+
+            survey.utils.showSurveyLink();
+
+            // Show the survey panel when the link is clicked
+            survey.select
+              .surveyLink()
+              .addEventListener("click", async () =>
+                popup.panel.update("survey")
+              );
+
+            survey.select
+              .closeSurveyLinkButton()
+              .addEventListener("click", async () =>
+                survey.utils.closeSurveyLink()
```

Comment:
UX Suggestion: I recommend closing the panel after running this click. See `window.close()` elsewhere [in this file](https://github.com/mozilla/fx-private-relay-add-on/blob/main/src/js/popup/popup.js#L66).

Reply:
I tried it out and didn't like it. If you click on the 'x', you'd probably expect the survey banner (purple banner) to go away, not the whole window. I think I'll keep it as is, but definitely do `window.close` for when the user clicks to take the survey using the external link.